<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="cube">

    <xacro:property name="mass_cube" value="0.5"/>
    <xacro:property name="dimension_cube" value="0.3"/>

    <link name="world"/>

    <joint name="odom_to_world" type="fixed">
        <parent link="world"/>
        <child link="odom_button"/>
        <origin rpy="0 0 0" xyz="2.0 0.0 1.0"/>
    </joint>

    <link name="odom_button"/>

    <joint name="articulation_joint" type="prismatic">
        <parent link="odom_button"/>
        <child link="button_x"/>
        <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
        <axis xyz="1 0 0"/>
        <limit effort="1000.0" velocity="5.0" lower="-100" upper="100"/>
        <dynamics spring_mount="0 0 0" stiffness="10.0" damping="1"/>
    </joint>

    <link name="button_x">
        <inertial>
            <mass value="${mass_cube}"/>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <inertia ixx="0.12" ixy="0" ixz="0" iyy="0.14" iyz="0" izz="0.18"/>
        </inertial>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0.0"/>
            <geometry>
                <box size="${dimension_cube} ${dimension_cube} ${dimension_cube}"/>
            </geometry>
            <material name="">
                <contact name="rubber"/>
            </material>
        </collision>
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <geometry>
                <box size="${dimension_cube} ${dimension_cube} ${dimension_cube}"/>
            </geometry>
            <material name="">
                <color rgba="1 1 1 1"/>
            </material>
        </visual>
    </link>
</robot>
